src/scripts/candidate_calc_outside_spending.js

```diff
@@ -0,0 +1,206 @@
+          if ( transactionTypes.includes( record['Form_Type'] ) ) {            
+
+            const { Tran_Amt1, Cand_NamL, Cand_NamF, Sup_Opp_Cd } = record;
+
+            // The Candidate names in Cand_NamL, Cand_NamF are not constant. 
+            // For some the name is only in Cand_NamL for others it is split between Cand_NamL and Cand_NamF. 
+            // This assignment makes them consistent.
+            let Candidate_Full_Name = Cand_NamF === '' ? Cand_NamL : `${Cand_NamF} ${Cand_NamL}`;
+
+            return { Tran_Amt1, Candidate_Full_Name, Sup_Opp_Cd }; 
+          }
+        }
+      })
+    );
+
+    try {
+      for await (const record of parser) {
+        output.push(record);
+      }
+    } catch(e) {
+      console.log(e);
+    }
+
+  }
+
+  return output;
+}
+
+
+/**
+ * @typedef {object} CandidateSpending
+ * @property {string} candidateName - Name of the Candidate as 'First Last'
+ * @property {string} office - The office the Candidate is running for
+ * @property {string} year - The election year that the Candidate running in
+ * @property {number} supportSum - Decimal number for the sum of all spending in support of a candidate
+ * @property {number} opposedSum - Decimal number for the sum of all spending in opposition to a candidate
+ */
+
+/**
+ * Reads a list of candidates and a list of transactions. Returns the total spent on supporting or opposing each candidate.
+ * @param {string[]} candidateNames - 
+ * @param {Transactions[]} transactions - 
+ * @returns {CandidateSpending[]}
+ */
+// function getSpendingAmounts(candidateNames, transactions) {
+function getSpendingAmounts(candidateInformation, transactions) {
+
+  let candidateSums = candidateInformation.map( candidate =>  {
+
+    let supportSum = transactions
+    .filter( (transaction) => candidate.Candidate_Name  === transaction.Candidate_Full_Name)
+    .filter( (transaction) =>  transaction.Sup_Opp_Cd === 'S')
+    .reduce( (accumulator, currentValue) => accumulator + Number.parseFloat(currentValue.Tran_Amt1), 0);
+
+    let opposedSum = transactions
+    .filter( (transaction) => candidate.Candidate_Name  === transaction.Candidate_Full_Name)
+    .filter( (transaction) =>  transaction.Sup_Opp_Cd === 'O')
+    .reduce( (accumulator, currentValue) => accumulator + Number.parseFloat(currentValue.Tran_Amt1), 0);
+
+    return {
+      candidateName: candidate.Candidate_Name,
+      office: candidate.Office,
+      year: candidate.Year,
+      supportSum,
+      opposedSum
+    };
+    
+  });
+
+  return candidateSums;
+}
+
+
+/**
+ * For each candidate this writes the total outside spending to support and oppose them.  
+ * @param {CandidateSpending[]} outsideSpending 
+ */
+function saveOutsideSpendingToJSON(outsideSpending) {
+  
+  for (const candidate of outsideSpending) {
+
+    // Replace all spaces in candidate names with underscores '_'
+    const candidatePathName = candidate.candidateName.split(' ').join('_').toLowerCase();
```

Comment:
Would it be better to use candidate.replaceAll(' ', '_').toLowerCase();

Reply:
String.prototype.replaceAll() does not appear to be currently supported in Node.js.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replaceAll#Browser_compatibility